### DISCUSSION TITLE
Exit Code 1 / callCatchAppNotFound

### DISCUSSION THREAD

**Discussion by akaneapple** [aclap-dev/video-downloadhelper#1019] [Help and Q&A]
### Your question

I've been having increasing trouble with Video DownloadHelper lately on a number of different sites.

I've tried both the new interface and the legacy UI. 

When I revert to the legacy UI, this is the "Exit code: 1"  error I receive:

```
moz-extension://d055a1e4-81cd-4ad0-a4fd-55249a026c4a/background/main.js:1
receive@moz-extension://d055a1e4-81cd-4ad0-a4fd-55249a026c4a/background/main.js:1:13152
gd</callCatchAppNotFound/</<@moz-extension://d055a1e4-81cd-4ad0-a4fd-55249a026c4a/background/main.js:1:42657
```

Here's my settings info:

```
version: 9.0.2.14a1
target: mozilla
channel: beta
lang: en-US
coapp: {"found":true,"path":"/Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp","version":"2.0.19","new_version":false}
license: {"accepted":true,"email":"<removed>","key":"removed"}
platform: x86-64 mac
UA: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:128.0) Gecko/20100101 Firefox/128.0
{
    "valid_license_message_has_been_displayed": true,
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "open_count_store": 1
}
```

As an example, I'm trying to download a video from [Vimeo](https://vimeo.com/channels/staffpicks/73234721). I receive the error. I've tried JDownloader2 as an alternative, and it works fine. I'm trying to get to the bottom of why VDH is failing. If anyone has any thoughts, please help.

I did look for similar discussions and found some related to "Exit Code 1," but the solution wasn't exactly clear and didn't seem to be an exact replica of my issue.

### Addon & CoApp version and details

version: 9.0.2.14a1
target: mozilla
channel: beta
lang: en-US
coapp: {"found":true,"path":"/Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp","version":"2.0.19","new_version":false}
license: {"accepted":true,"email":"<removed>","key":"removed"}
platform: x86-64 mac
UA: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:128.0) Gecko/20100101 Firefox/128.0
{
    "valid_license_message_has_been_displayed": true,
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "open_count_store": 2
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

I'm running Windows 11 Pro 64-bit version 23H2 build 22631.3880, Firefox 128.0 64-bit, licensed VDH 9.0.2.14a1 beta, CoApp 2.0.19.  That's pretty much the same as you except for the operating system.

I visited your Vimeo example & VDH downloaded it without a problem.  You might want to check this discussion: #1015.  It seems Vimeo itself might have been having trouble.  It certainly seems to be fine now.

**Reply by akaneapple** [reply]
Thanks for the response. I used Vimeo as an example, but it's not really a Vimeo issue. I should have added that this problem seems to persist with anything that uses HLS streaming. I just tried it on YouTube and Kickstarter (with similar HLS streaming videos) and received the same "Exit Code 1" error.

Additionally, the Companion App seems to be giving the same "Exit Code 1" error when trying to download a non-streaming video. I tried to download a WEBM video from the main page of Wikimedia Commons, and it resulted in an error. However, when using the Browser instead of the Companion App, I am able to download it normally through FF.

So I pretty much cannot download any visit except 1) if it's non-HLS-streaming and 2) using the "Browser" option instead of the "Companion App" option.

The only other thing I can think of is that perhaps another FF extension is interfering. I'll experiment some more and see if I can figure anything else out on my end.

**Reply by akaneapple** [reply]
For some reason my previous reply in mid-July triggered some sort of false flag ban and my account was banned out for 2.5 months for this thread. A support ticket was finally responded to and I was unbanned (it was indeed a mistake), so I'm politely  bumping this again.

**Reply by GrampaWildWilly** [reply]
Since July, my environment has undergone a few changes: Windows 11 Pro 64-bit version 24H2 build 26100.1882, Firefox 131.0 64-bit, licensed VDH 9.1.0.41 beta, CoApp 2.0.20.  Your Vimeo video still worked fine for me.  Downloaded fine.  Played fine.  No incidents.

**Reply by Mattmjs** [reply]
@akaneapple , Have you checked the companion app is not blocked by the anti virus & firewall ? It can be flagged as a virus etc..
but it's a false positive. That's what it sounds like to me if browser downloading works.

**Comment by akaneapple** [comment]
I just thought I would post to this thread since I'm no longer having issues. I honestly think being on an older version of OSX (Catalina at the time) was the culprit. I'm now on Sequoia (same laptop), still using Firefox, still using the same FF Add-Ons... 

Just in case anyone stumbles upon this thread and is using an older version of OSX: just know that might be the culprit.

**Reply by judovana** [reply]
I doubt a bit. I'm having it randomly on some vides. In same session, some works, some dont.  Various  privae windows, clean sessions and so on, simply do not work with this error, soem keep working.
